src/widgets/spinbox/lv_spinbox.c

```diff
@@ -175,10 +225,25 @@ void lv_spinbox_set_max_value(lv_obj_t * obj, int32_t max_value)
     LV_ASSERT_OBJ(obj, MY_CLASS);
     lv_spinbox_t * spinbox = (lv_spinbox_t *)obj;
 
-    spinbox->range_max = max_value;
+    /* If LV_SPINBOX_MAX_DIGIT_COUNT is ever < 10, then this
+     * value must be clamped to stay within that limit.  As long as
+     * LV_SPINBOX_MAX_DIGIT_COUNT == 10, this code will be optimized away. */
+    if(LV_SPINBOX_MAX_DIGIT_COUNT < 10) {
+        const int64_t max64 = lv_pow(10, LV_SPINBOX_MAX_DIGIT_COUNT) - 1;
+        const int32_t max = (int32_t)max64;
```

Comment:
To avoid a lot of casting I'd add a function like 
```c
static int32_t get_limit(void) 
{
    return (int32_t) lv_pow(10, LV_SPINBOX_MAX_DIGIT_COUNT) - 1;
}
```
```suggestion
        const int32_t max = get_limit();
```

Reply:
As it turns out, it was decided the simpler solution was preferred, so this code won't apply.  The actual function of `LV_SPINBOX_MAX_DIGIT_COUNT` will thus remain for documentation only.